pkg/apply/applier_test.go

```diff
@@ -390,75 +372,55 @@ var clusterScopedObjInfo = &resource.Info{
 	},
 }
 
-var clusterScopedObj2Info = &resource.Info{
-	Name: "cluster-scoped-2",
-	Mapping: &meta.RESTMapping{
-		Scope: meta.RESTScopeRoot,
-	},
-	Object: &unstructured.Unstructured{
-		Object: map[string]interface{}{
-			"apiVersion": "rbac.authorization.k8s.io/v1",
-			"kind":       "ClusterRoleBinding",
-			"metadata": map[string]interface{}{
-				"name": "cluster-scoped-2",
+func createNamespaceInfo(ns string) *resource.Info {
+	return &resource.Info{
+		Name: ns,
+		Object: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Namespace",
+				"metadata": map[string]interface{}{
+					"name": ns,
+				},
 			},
 		},
-	},
+	}
 }
 
 func TestValidateNamespace(t *testing.T) {
 	tests := map[string]struct {
 		objects []*resource.Info
-		isValid bool
+		err     error
 	}{
 		"No resources is valid": {
 			objects: []*resource.Info{},
-			isValid: true,
+			err:     nil,
 		},
-		"One resource is valid": {
-			objects: []*resource.Info{obj1Info},
-			isValid: true,
-		},
-		"Two resources with same namespace is valid": {
+		"Resources other than namespace is valid": {
 			objects: []*resource.Info{obj1Info, obj2Info},
-			isValid: true,
-		},
-		"Two resources with same namespace and cluster-scoped obj is valid": {
-			objects: []*resource.Info{obj1Info, clusterScopedObjInfo, obj2Info},
-			isValid: true,
+			err:     nil,
 		},
-		"Single cluster-scoped obj is valid": {
-			objects: []*resource.Info{clusterScopedObjInfo},
-			isValid: true,
+		"Different namespace resource is valid": {
+			objects: []*resource.Info{createNamespaceInfo("foo")},
+			err:     nil,
 		},
-		"Multiple cluster-scoped objs is valid": {
-			objects: []*resource.Info{clusterScopedObjInfo, clusterScopedObj2Info},
-			isValid: true,
-		},
-		"Two resources with differing namespaces is not valid": {
```

Comment:
So with this change, there can be resource A in namespace foo and resource B in namespace bar can be tracked/applied as part of single inventory object in namespace foo ?

Reply:
Yeah